incubator/crate/templates/statefulset.yaml

```
@@ -0,0 +1,98 @@
+apiVersion: "apps/v1"
+kind: StatefulSet
+metadata:
+  name: crate
+spec:
+  serviceName: "crate-db"
+  replicas: {{ .Values.crate.replicas }}
+  selector:
+    matchLabels:
+      app: {{ .Values.app }}
+  template:
+    metadata:
+      labels:
+        app: {{ .Values.app }}
+        app.kubernetes.io/name: crate-deployment
+        app.kubernetes.io/part-of: crate-db
+        app.kubernetes.io/version: {{ .Values.image.tag }}
+        app.kubernetes.io/managed-by: helm
+      annotations:
+        pod.alpha.kubernetes.io/initialized: "true"
+    spec:
+      initContainers:
+      - name: init-sysctl
+        image: busybox
+        imagePullPolicy: IfNotPresent
+        command: ["sysctl", "-w", "vm.max_map_count=262144"]
+        securityContext:
+          privileged: true
+      containers:
+      - name: crate
+        image: {{ .Values.image.name }}:{{ .Values.image.tag }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        command:
+          - /docker-entrypoint.sh
+          - -Clicense.enterprise=false
+          - -Ccluster.name={{ .Values.crate.clusterName }}
+          - -Cnetwork.host="0.0.0.0"
+          - -Cdiscover.seed_providers=srv
+          - -Cdiscovery.srv.query=_{{ .Values.service.clusterName }}._tcp.{{ .Values.service.name }}.{{ .Release.Namespace }}.svc.cluster.local
+          - -Cgateway.recover_after_nodes={{ .Values.crate.recoverAfterNodes }}
+          - -Cgateway.recover_after_time={{ .Values.crate.recoverAfterTime }}
+          - -Cgateway.expected_nodes={{ .Values.crate.replicas }}
+          - -Chttp.cors.enabled={{ .Values.http.cors.enabled }}
+          - -Chttp.cors.allow-origin={{ .Values.http.cors.allowOrigin }}
+          - -Cpath.data=/data
+          {{- if .Values.resources.limits.cpu }}
+          - -Cprocessors={{ .Values.resources.limits.cpu }}
+          {{- end }}
+        volumeMounts:
+            - mountPath: /data
+              name: data
+        resources:
+{{ toYaml .Values.resources | indent 10 }}
+        ports:
+        - containerPort: {{ .Values.service.adminPort }}
+          name: {{ .Values.service.adminName }}
+        - containerPort: {{ .Values.service.clusterPort }}
+          name: {{ .Values.service.clusterName }}
+        - containerPort: {{ .Values.service.postgresPort }}
+          name: {{ .Values.service.postgresName }}
+        env:
+        - name: CRATE_HEAP_SIZE
+          value: {{ .Values.crate.crateHeapSize | quote }}
+        - name: EXPECTED_NODES
+          value: {{ .Values.crate.replicas | quote }}
+        - name: CLUSTER_NAME
+          value: {{ .Values.crate.clusterName | quote }}
+        - name: RECOVER_AFTER_NODES
+          value: {{ .Values.crate.recoverAfterNodes | quote }}
+        livenessProbe:
```

The discovery requires to know the name of the CrateDB nodes, so we use the pod name as node name (usage see comment above):

```suggestion
        - name: POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        livenessProbe:
```